### PR TITLE
Add ability to interrupt kernel

### LINKED ISSUE #405: Ability to interrupt kernel
We only have a way to kill the kernel currently. We also need an "interrupt kernel" menu item.

### PR DESCRIPTION
**Pull request by captainsafia**
Closes #405.

Gotta figure out Windows support....

### CODE REVIEW

**Comment by captainsafia** [comment]
Hard to add Windows support when you don't have a Windows machine — so I just set it to error on  Windows for now.

Test for this is very meh. I'm not entirely sure if there is a way to check if a process has been interrupted, at least a way that is evident in `state.app.spawn`.

**Comment by rgbkrk** [comment]
It's ok about the uncertainty with Windows. What does jupyter/notebook do?

File: src/notebook/reducers/app.js [hunk 1]
```diff
@@ -37,6 +37,14 @@ export default handleActions({
     return cleanupKernel(state);
   },
   [constants.KILL_KERNEL]: cleanupKernel,
+  [constants.INTERRUPT_KERNEL]: function interruptKernel(state) {
+    if (process.platform === 'win32') {
+      console.error('Windows support for kernel interrupt not avaiable yet.');
```
Comment (rgbkrk) [inline src/notebook/reducers/app.js:42]: This should probably be a more user-visible error, using the notification system.

**Comment by captainsafia** [comment]
`jupyter/notebook` uses [its own Windows interrupt module](https://github.com/jupyter/jupyter_client/blob/02f9c580f6533ba97d1fed7157bef80c69f0f4ed/jupyter_client/win_interrupt.py).

File: src/notebook/reducers/app.js (hunk 1 above)
Comment (captainsafia) [inline src/notebook/reducers/app.js:42]: Oops!

**Pull request merged**